File: backend-java/src/main/java/com/kwikpic/model/Group.java
package com.kwikpic.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

/**
 * Maps to Mongoose Group schema in models/Group.js.
 * Central hub connecting Photographer → Photos → Participants.
 */
@Document(collection = "groups")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Group {

    @Id
    private String id;

    private String name;

    /** Unique 6-digit event code clients use to join. */
    @Indexed(unique = true)
    private String code;

    /** ObjectId hex-string referencing the owning Photographer. */
    private String photographer;

    /** ObjectId hex-strings referencing Photo documents. */
    private List<String> photos = new ArrayList<>();

    /** ObjectId hex-strings referencing User (participant) documents. */
    private List<String> participants = new ArrayList<>();

    private Date createdAt;
}

File: backend/src/models/Group.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const GroupSchema = new Schema({
  name: String,
  code: { type: String, required: true, unique: true }, // 6-digit Unique Code
  
  // Link to the Photographer
  photographer: { type: Schema.Types.ObjectId, ref: 'Photographer' },
  
  // Array of Photo IDs
  photos: [{ type: Schema.Types.ObjectId, ref: 'Photo' }],

  // --- NEW FIELD (REQUIRED FOR JOINING) ---
  // This stores the IDs of Users (Clients) who join the group
  participants: [{ type: Schema.Types.ObjectId, ref: 'User' }], 
  // ----------------------------------------

  createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Group', GroupSchema);